tests/testthat/test_class_phg_metrics.R

```diff
@@ -0,0 +1,348 @@
+        "valid_algn_from_path" = algnPath,
+        "valid_algn_from_df"   = algnDf
+    )
+    expect_true(is(metricsTable(metTest), "list"))
+    expect_equal(
+        object   = metricsIds(metTest),
+        expected = c(
+            "toy_anchors_s01",
+            "valid_algn_from_path",
+            "valid_algn_from_df",
+            "toy_gvcf_metrics",
+            "valid_gvcf_from_path",
+            "valid_gvcf_from_df"
+        )
+    )
+
+    metTest <- metBase
+    metricsTable(metTest) <- list(
+        "valid_gvcf_from_path"    = gvcfPath,
+        "valid_gvcf_from_df"      = gvcfDf,
+        "invalid_gvcf_from_path"  = "invalid_path",
+        "invalid_gvcf_from_path2" = hvcfPath,
+        "invalid_gvcf_from_df"    = mtcars,
+        "valid_algn_from_path"    = algnPath,
+        "valid_algn_from_df"      = algnDf,
+        "invalid_algn_from_path"  = "invalid_path",
+        "invalid_algn_from_path2" = hvcfPath,
+        "invalid_algn_from_df"    = mtcars
+    )
+    expect_true(is(metricsTable(metTest), "list"))
+    expect_true(is(metricsTable(metTest, type = "gvcf"), "list"))
+    expect_true(is(metricsTable(metTest, type = "align"), "list"))
+    expect_equal(
+        object   = metricsIds(metTest),
+        expected = c(
+            "toy_anchors_s01",
+            "valid_algn_from_path",
+            "valid_algn_from_df",
+            "toy_gvcf_metrics",
+            "valid_gvcf_from_path",
+            "valid_gvcf_from_df"
+        )
+    )
+    metOut01 <- utils::capture.output(metTest)
+    expect_equal(length(metOut01), 9)
+
+    metTest <- metBase
+    metricsTable(metTest) <- list(
+        "toy_anchors_s01"      = algnDf,
+        "valid_algn_from_path" = algnPath
+    )
+    expect_equal(
+        object   = metricsIds(metTest),
+        expected = c(
+            "toy_anchors_s01",
+            "valid_algn_from_path",
+            "toy_gvcf_metrics"
+        )
+    )
+
+})
+
+
+test_that("PHGMetrics general dot plot tests", {
+    metricDir <- system.file("extdata", package = "rPHG2")
+
+    metBase <- PHGMetrics(metricDir)
+
+    expect_true(is(plotDot(metBase, metBase$toy_anchors_s01), "ggplot"))
+    expect_true(is(plotDot(metBase, metBase$toy_anchors_s01, colorId = "score"), "ggplot"))
+    expect_true(is(plotDot(metBase, metBase$toy_anchors_s01, refSeqId = c("1", "2")), "ggplot"))
+    expect_true(is(plotDot(metBase, metBase$toy_anchors_s01, querySeqId = c("1", "2")), "ggplot"))
+    expect_error(plotDot(metBase, metBase$toy_anchors_s01, colordId = "not_a_column"))
+    expect_error(plotDot(metBase, metBase$toy_anchors_s01, refSeqId = "1", querySeqId = "2"))
+    expect_error(plotDot(metBase, metBase$toy_gvcf_metrics))
+    expect_error(
+        plotDot(
+            object = metBase,
+            metricId = c(
+                metBase$toy_anchors_s01,
+                metBase$toy_anchors_s01
+            ),
+            refSeqId = "1",
+            querySeqId = "2"
+        )
+    )
+
+})
+
```

Comment:
Not that it really matters, but is there a reason for all these extra new lines?

Reply:
Fixed. I usually add a big buffer of new lines when typing/editing so I can constantly have text lines near the top-middle part of my screen, and I forgot to remove them.